xxx-uv_link_t.md

```diff
@@ -0,0 +1,73 @@
+| Title  | Port core to uv_link_t      |
+|--------|-----------------------------|
+| Author | @indutny                    |
+| Status | DRAFT                       |
+| Date   | 2016-06-03                  |
+
+## 1. Proposal
+
+I propose to replace `StreamBase` with [`uv_link_t`][0], and port existing
+classes that inherit from `StreamBase`.
+
+## 2. Current state of C++ Streams in Node
+
+The fast HTTP and TLS protocol implementation in Node core depends on so called
+`StreamBase` C++ class. This class is an analog of JavaScript streams in C++.
+The main ideas behind `StreamBase` are to:
+
+1. Avoid unnecessary memory allocations by reading data directly into buffer
+   from which it will be synchronously parsed
+2. Avoid calling JavaScript unless absolutely necessary
+
+However, this API has lots of dependencies on the core internals, and thus
+can't be used outside of it.
+
+## 3. uv_link_t
+
+From [`uv_link_t`][0] readme:
+
+    Chainable libuv streams.
+
+    It is quite easy to write a TCP server/client in [libuv][1]. Writing HTTP
+    server/client is a bit harder. Writing HTTP server/client on top of TLS
+    server could be unwieldy.
+
+    `uv_link_t` aims to solve complexity problem that quickly escalates once
+    using multiple layers of protocols in [libuv][1] by providing a way to
+    implement protocols separately and chain them together in easy and
+    high-performant way using very narrow interfaces.
+
+Given that [`uv_link_t`][0] depends only on [libuv][1], it is very easy to
+envision how it will be used in Node.js addons. Even core modules will no longer
+be required to be distributed with core, since they could be built without core
```

Comment:
I have a hint of a concern that people will take this as meaning core modules will be distributed without core. Possibly change the wording to simply say that it'd be easier to write alternative implementations of core modules (e.g. alternative http implementations). Only suggest this to keep the proposal closer to core today (as an easier way for this EP to get accepted).

note: i'm not suggesting that i wouldn't like to distribute the core modules separately in the future. :)